dsp/transform/hilbert.go

```diff
@@ -0,0 +1,83 @@
+// Copyright ©2024 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"gonum.org/v1/gonum/dsp/fourier"
+)
+
+// Hilbert implements an approximate Hilbert transform that allows calculation
+// of an approximate analytical signal of a real signal, and determine the
+// real envelope of a signal.
+type Hilbert struct {
+	fft  *fourier.CmplxFFT
+	work []complex128
+}
+
+// NewHilbert returns a new Hilbert transformer for signals of size n.
+// The transform is most efficient when n is a product of small primes.
+func NewHilbert(n int) *Hilbert {
+	if n < 1 {
+		return &Hilbert{nil, nil}
+	}
```

Comment:
Please remove this and allow a panic. A negative would be indicative of failure to hold a known invariant.